src/components/common/field-selector.js

```diff
@@ -94,16 +103,27 @@ class FieldSelector extends Component {
   };
 
   fieldsSelector = props => props.fields;
+  filteredFieldsSelector = props =>
+    props.fields.filter(field => !(props.value || []).includes(field.name));
   valueSelector = props => props.value;
   filterFieldTypesSelector = props => props.filterFieldTypes;
   showTokenSelector = props => props.showToken;
 
   selectedItemsSelector = createSelector(this.fieldsSelector, this.valueSelector, (fields, value) =>
-    fields.filter(f => (Array.isArray(value) ? value : [value]).includes(defaultDisplayOption(f)))
+    fields.filter(f =>
+      Boolean(
+        (Array.isArray(value) ? value : [value]).find(d => {
```

Comment:
You can use `toArray` from utils/utils

use `notNullorUndefined(d)` from 'utils/data-utils' instead of `(!d)`